### ISSUE TITLE
I need  some advices about training model

### ISSUE THREAD

**Issue by coderclear** [matterport/Mask_RCNN#652]
I have 4200+ images to train model.  And configuration is 
Configurations:
BACKBONE                       resnet101
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     2
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0
DETECTION_NMS_THRESHOLD        0.3
GPU_COUNT                      1
GRADIENT_CLIP_NORM             5.0
IMAGES_PER_GPU                 2
IMAGE_MAX_DIM                  512
IMAGE_META_SIZE                14
IMAGE_MIN_DIM                  512
IMAGE_MIN_SCALE                2.0
IMAGE_RESIZE_MODE              crop
IMAGE_SHAPE                    [512 512   3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'mrcnn_mask_loss': 1.0, 'rpn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [123.7 116.8 103.9]
MINI_MASK_SHAPE                (56, 56)
NAME                           davis
NUM_CLASSES                    2
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        2000
POST_NMS_ROIS_TRAINING         1000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (8, 16, 32, 64, 128)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.8
RPN_TRAIN_ANCHORS_PER_IMAGE    64
STEPS_PER_EPOCH                2104
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           128
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.001

train coding is:
model = modellib.MaskRCNN(mode="training", config=config,
                          model_dir=MODEL_DIR)
init_with = "coco"  # imagenet, coco, or last
model.load_weights(COCO_MODEL_PATH, by_name=True,
                       exclude=["mrcnn_class_logits", "mrcnn_bbox_fc", 
                                "mrcnn_bbox", "mrcnn_mask"])

model.train(dataset_train, dataset_val, 
            learning_rate=config.LEARNING_RATE, 
            epochs=15, 
            layers='heads')
model.train(dataset_train, dataset_val, 
            learning_rate=config.LEARNING_RATE / 10,
            epochs=15, 
            layers="all")
loss is 

![image](https://user-images.githubusercontent.com/18438341/41088549-5f7e5c36-6a72-11e8-9fe2-5917e3fe1d3b.png)

![image](https://user-images.githubusercontent.com/18438341/41088576-777f48f4-6a72-11e8-89cf-9eaec52c2aef.png)

Thanks for all your response!

**Comment by zungam** [comment]
Some questionts.
How many classes do you use? That is, how many type of objects are you training on?

You training loss is decrasing but your out of sample loss is increasing. It means that there is an overfit to your training set from the start. Your model works good on your training dataset, but is getting worse on your out of sample training set. Have you split your trainingset into validation set and training set? Is the validation set seperate images? If so, how different are the validation set from the training set? Are they completly similar in style and domain?

**Comment by coderclear** [comment]
@zungam  Thanks for your response. I know this is overfitted, but I don't know why. I have training set and validation set, this two have no intersection. I think they're similar in style and domain.  Specifically, you can refer to this https://davischallenge.org/
I had tried to change learning rate to 0.0001, but no work.
Thank you!

Below are all graphs:
[lossgraph.pdf](https://github.com/matterport/Mask_RCNN/files/2080634/lossgraph.pdf)

**Comment by zungam** [comment]
Hmm. Very weird behavour indeed. I dont have any clue what this could be, but I doubt there is something wrong with maskrcnn, so I would look into how you load the training set and validation set, check if there is any difference in your code in the load dataset function

**Comment by lijianaiml** [comment]
i think your training process may be wrong.In your trian code ,if you want use epoch 15 in first training stage and epoch 15 in second training,you should write like this.

model.train(dataset_train, dataset_val,
learning_rate=config.LEARNING_RATE,
epochs=15,
layers='heads')
model.train(dataset_train, dataset_val,
learning_rate=config.LEARNING_RATE / 10,
epochs=30,
layers="all")
